plugins/optimization-detective/class-od-url-metric-group.php

```diff
@@ -233,7 +235,12 @@ public function is_complete(): bool {
 			}
 			$current_time = microtime( true );
 			foreach ( $this->url_metrics as $url_metric ) {
-				if ( $current_time > $url_metric->get_timestamp() + $this->freshness_ttl ) {
+				if (
+					$current_time > $url_metric->get_timestamp() + $this->freshness_ttl
+					||
+					// If the URL Metric's ETag doesn't match the current ETag, consider the URL metric as stale.
+					( null !== $this->collection && $url_metric->get_etag() !== $this->collection->get_current_etag() )
```

Comment:
Had to write it this way because `$collection` is optional and defaults to `null`.